src/artemis/fast_grid_scan_plan.py

```diff
@@ -84,8 +88,9 @@ def do_fgs():
         yield from bps.kickoff(fgs_motors)
         yield from bps.complete(fgs_motors, wait=True)
 
-    with NexusWriter(parameters):
-        yield from do_fgs()
+    with NexusWriter(create_parameters_for_first_file(parameters)):
+        with NexusWriter(create_parameters_for_second_file(parameters)):
+            yield from do_fgs()
```

Comment:
This double indentation is a little awkward, and I'm not a big fan of the plan calling both file writers separately (it makes them harder to split up when we want to do 2D scans, and doesn't work well with the potential future implementation of the ispyb writing in this PR https://github.com/DiamondLightSource/python-artemis/pull/186 )

Maybe a wrapper/superclass that calls both filewriters, so we only need to call the nexuswriter from the plan once, and it can then write 1 or 2 files as appropriate?

Reply:
FYI you _can_ [write multiple items in a single `with` statement](https://docs.python.org/3/reference/compound_stmts.html#the-with-statement), e.g.:
```
with A() as a, B() as b:
    # code
```
or
```
with (
    A() as a,
    B() as b,
):
    # code
```

Reply:
I've changed to the multiple with statements, I don't like it but I think a larger refactor should be addressed in https://github.com/DiamondLightSource/python-artemis/issues/164